components/user/Tabs/SettingsContent.tsx

```diff
@@ -1,59 +1,237 @@
+  user,
+  authedUser,
+  toast,
+  reloadUser,
+  onTabChange,
+}: Props) {
+  const [$error, $setError] = useState<string>(UNSET)
+  const [$userData, $setUserData] = useState<API.ApiUserMetadata | null>(
+    authedUser
+  )
+
+  const {
+    email: _email = UNSET,
+    graffiti: _graffiti = UNSET,
+    discord: _discord = UNSET,
+    telegram: _telegram = UNSET,
+    country_code: _cc = UNSET,
+  } = $userData || {}
+  const $graffiti = useField({
+    ...EDITABLE_FIELDS.graffiti,
+    defaultValue: _graffiti,
+  })
+
+  const $email = useField({
+    ...EDITABLE_FIELDS.email,
+    defaultValue: _email,
+  })
+  const $discord = useField({
+    ...EDITABLE_FIELDS.discord,
+    defaultValue: _discord,
+    touched: !!_discord,
+  })
+  const $telegram = useField({
+    ...EDITABLE_FIELDS.telegram,
+    defaultValue: _telegram,
+    touched: !!_telegram,
+  })
 
-  const { choice: socialChoice, value: socialValue } =
-    getSocialChoice(authedUser)
+  const $country = useField({
+    ...EDITABLE_FIELDS.country,
+    defaultValue: _cc,
+  })
+  const testInvalid = useCallback(() => {
+    const invalid =
+      !$email?.valid ||
+      !$graffiti?.valid ||
+      !$discord?.valid ||
+      !$telegram?.valid ||
+      !$country?.valid
+    if (invalid) {
+      $setError('Please correct the invalid fields below')
+      scrollUp()
+    } else {
+      $setError(UNSET)
+    }
+    return invalid
+  }, [$email, $graffiti, $telegram, $discord, $country])
+  const update = useCallback(async () => {
+    if (
+      !$email ||
+      !$graffiti ||
+      !$telegram ||
+      !$discord ||
+      !$country ||
+      !authedUser ||
+      testInvalid()
+    ) {
+      return
+    }
+    const email = $email?.value
+    const graffiti = $graffiti?.value
+    const telegram = $telegram?.value
+    const discord = $discord?.value
+    const country = $country?.value
 
+    const updates = {
+      email,
+      graffiti,
+      telegram,
+      discord,
+      country_code: country,
+    }
+
+    const result = await API.updateUser(authedUser.id, updates)
+
+    if ('error' in result) {
+      const error = '' + result.message
+      $setError(error)
+    } else {
+      toast.setMessage('User settings updated')
+      toast.show()
+      $graffiti.setTouched(false)
+      scrollUp()
+      // $setUser(result)
+      await reloadUser()
+    }
+  }, [
+    $email,
+    $graffiti,
+    $telegram,
+    $discord,
+    $country,
+    authedUser,
+    testInvalid,
+    toast,
+    reloadUser,
+  ])
+  useEffect(() => {
+    if (!authedUser) return
+    // local cache
+    $setUserData(authedUser)
+    const canSee =
+      authedUser &&
+      user &&
+      user.graffiti === authedUser.graffiti &&
+      user.id === authedUser.id
+    // eslint-disable-next-line no-console
+    if (!canSee) {
+      // if you try to go to /users/x/settings but you're not user x
+      onTabChange('weekly')
+      toast.setMessage('You are not authorized to go there')
```

Comment:
When I change graffiti and hit save, the page says "You are not authorized to go there" and redirects me to the weekly stats page. It also doesn't update the user page and tab title with my new graffiti

Reply:
It sounds like this will be addressed by reloading the page after the update finishes, right?